lazy-pages/src/sys/unix.rs

```diff
@@ -43,14 +47,22 @@ extern "C" fn handle_sigsegv(_sig: i32, info: *mut siginfo_t, ucontext: *mut c_v
         };
 
         let addr = (*info).si_addr();
-        let info = ExceptionInfo {
+        let exc_info = ExceptionInfo {
             fault_addr: addr as *mut _,
             is_write,
         };
 
-        super::user_signal_handler(info)
-            .map_err(|err| err.to_string())
-            .expect("Signal handler failed")
+        if let Err(err) = super::user_signal_handler(exc_info) {
+            let old_sig_handler_works = match err {
+                Error::SignalFromUnknownMemory {
+                    addr: _,
+                    wasm_mem_addr: _,
+                    wasm_mem_end_addr: _,
+                } => old_sig_handler(sig, info, ucontext),
+                _ => false,
+            };
+            assert!(old_sig_handler_works, "Signal handler failed: {}", err);
```

Comment:
why do we need production assertions there?

Reply:
Because if sig handler finish work with an error we have no ways to continue execution currently.

Reply:
> Because if sig handler finish work with an error we have no ways to continue execution currently.

So for unrecoverable errors I prefer use panic to have more concrete log

Reply:
Ok change to panic